### PR TITLE
modify track() to remove force unwraps

### PR DESCRIPTION
**Pull request by RamblinWreck77**
Minor change to remove force unwraps. Should work the exact same way, null event names will be "mp_event", and non-null non-empty event names will override var ev with the real name.

### CODE REVIEW

**Comment by RamblinWreck77** [comment]
Oops, forgot to check all schemes 🙄. Updating this now, and removing the unnecessary archive/reference to applicationWillTerminate that you mentioned @zihejia

**Comment by RamblinWreck77** [comment]
All I can do for now, will check back on these tests later

**Comment by zihejia** [comment]
hi @RamblinWreck77 , do you mind sharing your email address? We'd like to send you some Mixpanel Swag for your great help to us and the community along the way!

**Comment by Lordx3n0saeon** [comment]
> hi @RamblinWreck77 , do you mind sharing your email address? We'd like to send you some Mixpanel Swag for your great help to us and the community along the way!

Oh wow, that’s amazing and I’d love to. I’d rather not publicly dox my GitHub, can you post an @mixpanel email and I’ll reach out?

**Comment by zihejia** [comment]
> > hi @RamblinWreck77 , do you mind sharing your email address? We'd like to send you some Mixpanel Swag for your great help to us and the community along the way!
> 
> Oh wow, that's an amazing honor. Unfortunately I don't want to dox my email publicly. Would you mind replying to this message with an @mixpanel email? I'll send you a message and you can delete it as soon as I send it.

here is my email id: `zihe.jia` at mixpanel dot com

File: Sources/MixpanelInstance.swift [hunk 1]
```diff
@@ -434,61 +431,92 @@ open class MixpanelInstance: CustomDebugStringConvertible, FlushDelegate, AEDele
...
+                #endif
+
+                // If the system is calling the expiration handler, the taskID must be valid (?)
+                // Either way, there is no known harm in ending an invalid background task
+                // It is also possible iOS has some internal hack requiring a call to endBackgroundTask
+                // always, so by "defensively" calling endBackgroundTask here we are insuring iOS
+                // doesn't kill our process. Note that repeated "violations" of not ending your task
+                // will influence the background time iOS grants you in future requests.
+                sharedApplication.endBackgroundTask(self.taskId)
+                self.taskId = UIBackgroundTaskIdentifier.invalid
```
Comment (zihejia) [inline Sources/MixpanelInstance.swift:463]: I think you are spot on the issues for the background task. I will make the change based on your suggestion. Thanks!

**Comment by zihejia** [comment]
hi @RamblinWreck77 , I merged this [PR](https://github.com/mixpanel/mixpanel-swift/pull/491) which got your suggestions from here. So I'm closing this one now. As always, really appreciate your help!

**Pull request closed without merging**